project/src/main/webapp/emailParse.js

```diff
@@ -11,92 +11,87 @@
-    var exampleDict = {
-        "12345":{"date": "Some string", "body": "Your Microsoft interview will be held on October 2, 2019 at 01:45 PM Pacific Time (US & Canada)"},
-        "23456":{"date": "Some string", "body": "Arely Miranda González, a Product Manager at YouTube, started Karma Action, a new non-profit organization and webapp"},
-        "34567":{"date": "Some string", "body": "Your microsoft workshop is on Monday 4, 2019."},
-    };
-    var allPass = listOfPassages(exampleDict);
-    console.time("Using Promises Test");
-    const promises =allPass.map(
-        passage => model.findAnswers(question, passage),
-    );
-    Promise.all(promises).then((values) => {
-        var nonEmpty = ridOfEmptyArrays(values,allPass);
-        if (nonEmpty.size == 0)
-        {
-            console.log("No Answer Available");
+    extractEmailBodiestoArray(exampleDict) {
+        let allPass = [];
+        for (let key in exampleDict) {
+            let body = exampleDict[key]["emailBody"];
+            allPass.push(body.slice(300));
         }
-        else
-        {
-            var orderedConfidence = highestConfidence(nonEmpty);
-            var MLDictAnswer = nonEmpty.get(orderedConfidence);
-            console.log("original",values);
-            console.log("nonEmpty", nonEmpty);
-            console.log("highest confidence", orderedConfidence);
-            console.log("final answer: ", MLDictAnswer["answer"]);
-            var answer = document.getElementById("answer");
-            answer.innerHTML = MLDictAnswer["answer"];
-            console.timeEnd("Using Promises Test");
+        return allPass;
+    }
+    /** Keep emails that returned an answer from ML model */
+    ridOfEmptyArrays(mlValues, allPass) {
+        const confidenceWithEmailBody = new Map();
+        for (let i = 0; i < mlValues.length; i++) {
+            if (mlValues[i].length != 0) {
+                let mlAnswer = mlValues[i][0]["text"];
+                confidenceWithEmailBody.set(mlValues[i][0]["score"], { "answer": mlAnswer, "emailBody": allPass[i] })
+            }
         }
-    });   
-}
-
-function extractEmailBodiestoArray(exampleDict)
-{
-    var allPass = [];
-    for (var key in exampleDict)
-    {
-        allPass.push(exampleDict[key]["body"]);
+        return confidenceWithEmailBody;
     }
-    return allPass;
-}
 
-function ridOfEmptyArrays(MLvalues,allPass)
-{ 
-    let confidenceWithEmailBody = new Map();
-    for (let i = 0; i < MLvalues.length; i ++)
-    {
-        if (MLvalues[i].length != 0)
-        {  
-            MLAnswer = MLvalues[i][0]["text"];
-            confidenceWithEmailBody.set(MLvalues[i][0]["score"],{"answer":MLAnswer,"emailBody":allPass[i]})
+    highestConfidence(answerDict) {
+        let confidenceArray = [];
+        for (let key of answerDict.keys()) {
+            confidenceArray.push(key);
         }
+        let bestAnswer = confidenceArray.sort((a, b) => b - a);
+        return bestAnswer[0];
```

Comment:
Might want to check that bestAnswer isn't empty first:
```
return bestAnswer.length ? bestAnswer[0] : '';
```

Reply:
I check if the ML model return no answer in parseEmailsWithModel() (line 45). As a result, highConfidence() will only run if there is at least one answer in the parameter and will not return an empty answer.